madminer/delphes/delphes_reader.py

```diff
@@ -814,78 +859,6 @@ def _analyse_delphes_sample(
 
         return this_observations, this_weights, n_events
 
-    def _check_python_syntax(self, expression):
-        """
-        Evaluates a Python expression to check for syntax errors
-
-        Parameters
-        ----------
-        expression : str
-            Python expression to be evaluated. The evaluation raises either SyntaxError or NameError
-
-        Returns
-        -------
-            None
-        """
-
-        try:
-            eval(expression)
-        except SyntaxError:
-            raise ValueError("The provided Python expression is invalid")
-        except NameError:
-            pass
-
-    def _check_sample_observations(self, this_observations):
-        """ Sanity checks """
-        # Check number of events in observables
-        n_events = None
-        for key, obs in six.iteritems(this_observations):
-            this_n_events = len(obs)
-            if n_events is None:
-                n_events = this_n_events
-                logger.debug("Found %s events", n_events)
-
-            if this_n_events != n_events:
-                raise RuntimeError(
-                    "Mismatching number of events in Delphes observations for {}: {} vs {}".format(
-                        key, n_events, this_n_events
-                    )
-                )
-
-            if not np.issubdtype(obs.dtype, np.number):
-                logger.warning(
-                    "Observations for observable %s have non-numeric dtype %s. This usually means something "
-                    "is wrong in the definition of the observable. Data: %s",
-                    key,
-                    obs.dtype,
-                    obs,
-                )
-        return n_events
-
-    def _check_sample_weights(self, n_events, this_weights):
```

Comment:
What happened to this function, was it superfluous, or did you move the functionality somewhere else? Or am I just blind?

Reply:
Both `_check_sample_observations` and `_check_sample_weights` functions are identical but for what they check (_observation_ vs _weight_).

I created a generic function at the beginning of the class called `_check_sample_elements`, that replaces these two.

Although this is not a pure _"Drop Python2"_ change, I thought it was an _easy win_. Please, let me know if you think this generic function should have any other name.

Reply:
Good point, an easy win indeed!